roles/nova-data/tasks/libvirt.yml

```diff
@@ -56,29 +58,157 @@
   notify: restart nova services
   when: openstack_install_method == 'source'
 
+- name: make pki paths
+  file:
+    path: "{{ item }}"
+    state: directory
+    owner: root
+    group: kvm
+  with_items:
+    - /etc/pki/CA
+    - /etc/pki/libvirt/private
+
+- name: get uname -n
+  command: uname -n
+  register: uname
+
+- name: create 'uname -n' as fact 'uname_host'
+  set_fact:
+    uname_host: "{{ uname.stdout }}"
+
+- name: install CA file
+  template:
+    src: cacert.pem.j2
+    dest: /etc/pki/CA/cacert.pem
+    owner: root
+    group: kvm
+    mode: 0644
+  register: ca_install
+
+- name: upload CA key
+  template:
+    src: cacert.key.j2
+    dest: /tmp/cacert.key
+  register: ca_key_install
+
+- name: upload server certificate template
+  template:
+    src: server-cert-template.j2
+    dest: /tmp/server-cert-template
+  register: ca_template
+
+- name: create server cert private key
+  shell: "certtool --generate-privkey > /etc/pki/libvirt/private/serverkey.pem"
+  when: ca_install|changed or ca_key_install|changed or ca_template|changed
+
+- name: create server cert
+  command: "certtool --generate-certificate
+           --template /tmp/server-cert-template
+           --load-privkey /etc/pki/libvirt/private/serverkey.pem
+           --load-ca-certificate /etc/pki/CA/cacert.pem
+           --load-ca-privkey /tmp/cacert.key
+           --outfile /etc/pki/libvirt/servercert.pem"
+  when: ca_install|changed or ca_key_install|changed or ca_template|changed
+  notify: restart libvirt-bin
+
+- name: upload client certificate template
+  template:
+    src: client-cert-template.j2
+    dest: /tmp/client-cert-template
```

Comment:
Same as above; we delete them anyway. I think it's good to delete them rather than leave them around - I don't think they have any value in /etc/, and I figure /tmp/ is the best place for ephemeral stuff.